### PR TITLE
Backend functionality for allowing galleries for add textbox

### PR DESCRIPTION
**Pull request by changyang-liu**

### CODE REVIEW

File: django_project/galleries/views.py [hunk 1]
```diff
@@ -25,7 +30,7 @@ class GalleryRetrieve(generics.RetrieveAPIView):
     serializer_class = MainSiteGallerySerializer
 
 @api_view(["POST"])
-@login_required
+# @login_required
```
Comment (DumboOctopus) [inline django_project/galleries/views.py:33]: what was the purpose of removing login required?

File: django_project/galleries/serializers.py [hunk 2]
```diff
@@ -1,40 +1,102 @@
...
+    Combine ForeignKey related Image and GalleryTextField models into a shared
+    array using the 'index' field in both models to order them.
+    NOTE: This method is reliant on both models being ordered by 'index'
+    '''
+    image_data = img_serializer(obj.images.all(), many=True).data
+    textfield_data = text_serializer(obj.textfields.all(), many=True).data
+    data = []
+    # index for data, image_data, and textfield_data respectively
+    i, image_idx, text_idx = 0, 0, 0
+    while i < len(image_data) + len(textfield_data):
```
Comment (DumboOctopus) [inline django_project/galleries/serializers.py:53]: There is an error if the gallery ends with a text field.

**Review by DumboOctopus — CHANGES_REQUESTED** [review]
Looks great! Just some small things and edge cases  for the `get_images_and_text`. But I think even as it is now, I think its good enough to give to ethan for his frontend code.


Checks:
1. mainsite endpoint is still the same
2. Couple of edge cases tests for the mainsite endpoint

File: django_project/galleries/views.py (hunk 1 above)
Comment (changyang-liu) [reply django_project/galleries/views.py]: Oops! I removed it for debugging purposes but forgot to add it back in.

File: django_project/galleries/serializers.py (hunk 2 above)
Comment (changyang-liu) [reply django_project/galleries/serializers.py:53]: Oh right I forgot to add a check for the index out of range. I'll add it in

**Review by DumboOctopus — APPROVED** [review]
Looks good. I'll press merge
